test/meson.build

```diff
@@ -0,0 +1,38 @@
+gtest_dep = dependency('gtest', main : true, required : false)
+if not gtest_dep.found()
+    gtest_proj = import('cmake').subproject('googletest', required: false)
+    if gtest_proj.found()
+        gtest_dep = declare_dependency(
+            dependencies: [
+                dependency('threads'),
+                gtest_proj.dependency('gtest'),
+                gtest_proj.dependency('gtest_main'),
+            ]
+        )
+    else
+        assert(
+            not get_option('tests').enabled(),
+            'Googletest is required if tests are enabled'
+            )
+    endif
+endif
+
+testSourceFiles = [
+        'data_sync_config_test',
+    ]
+
+foreach testFile : testSourceFiles
+    test(
+        'test_' + testFile.underscorify(),
+        executable(
+            'test-' + testFile.underscorify(),
+            testFile + '.cpp',
+            rbmc_data_sync_sources,
+            dependencies : [
+                gtest_dep,
+                rbmc_data_sync_dependencies,
+            ],
+            include_directories: include_directories('../src'),
```

Comment:
I think we could use [inc_dir](https://github.com/ibm-openbmc/phosphor-data-sync/pull/4/commits/aee04830c645db9394d20f195059ca294373661a#diff-f28598af2e23aa5d2bc7c72e022ae2c56a33802eb970afffaeca1e40607f97feR23)? may be rename that variable into `src_inc_dir`?